### ISSUE TITLE
Shared memory not being created on macOS

### ISSUE THREAD

**Issue by JerryHyun** [eclipse-ecal/ecal#988]
### Problem Description

I am considering using eCAL for real-time synchronization of large-sized 3d mesh data. When I tested it on windows os using minimal_rec and minimal_snd samples, there was no problem. However, the test did not work on macOS, so I tried to find the cause of the problem.

1. macOS is very limited in the filename of shared memory. The maximum length for a filename currently is 31 characters per section (see PSHMNAMELENGTH in /usr/include/sys/posix_shm.h)
For this reason, the minimal_rec and minimal_snd samples are also problematic because the longest shared memory name created when running the minimal_rec sample is 38 characters.(/Hello_1058310900281_shm_38316_ack_evt)
And the default shutdown process memory name is also a problem because it is over 31 characters long.(/ecal_shutdown_process_38316_evt)
The solution is to use a random 4-character UNIQUE ID instead of the 13-character EPOCH TIME used to create the shared memory name to avoid duplicating the shared memory (if the reason for the EPOCH TIME is to avoid duplicates...).
**This change allows you to use topic names of up to 7 characters.**
Also, overriding EVENT_SHUTDOWN_PROC to "ecal_shutdown_proc" solves the shutdown process memory name issue.

2.  When setting g_process_name, if I give a length like std::string(buf, length); I get a problem on clang 13.0.0 x86_64-apple-darwin20.6.0.
Please see the code snippet below.
```
//strTest.cpp
#include <iostream>
#include <string.h>

int main()
{
	uint32_t length = 25;
	char buf[length];
	char origin[] = "/usr/temp";

	strncpy_s(buf, sizeof(buf), origin, sizeof(origin));
	std::string bufStr = std::string(buf, length);

	std::cout << bufStr << std::endl;

	return 0;
}
```
After building clang 13.0.0 x86_64-apple-darwin20.6.0, the results of the run are shown below.
```
jerry@Jerry-MacBookAir ~ %  ./strTest
/usr/temp�?l`QG�
jerry@Jerry-MacBookAir ~ % 
```
As you can see above, a strange value is added to the string.
The problem can be easily fixed as shown below.
`std::string(buf);`

3. There is an issue with not finding the ecaltime-localtime dynamic lib.
This can be easily fixed with a small modification to ecal_timegate.cpp.

Let's create a pull request to fix the above issue I found.

### How to reproduce

.

### How did you get eCAL?

Custom Build / Built from source

### Environment

- eCAL : 5.11.2
- OS : macOS Big Sur(v11.5.2)
- CMake : 3.25.2
- clang : 13.0.0 x86_64-apple-darwin20.6.0

**Comment by KerstinKeller** [comment]
Hi @JerryHyun,

thank you for reporting this and especially looking into this and opening a PR! We have discussed internally how we can improve MacOs support, as none of us are Mac Developers.

First, I'd suggest to create different PRs for the three different topics.

1) 31 character limit sounds like something which Windows would do 😄 Hashing the values to create unique IDs sounds pretty reasonable. However we will be a bit hesitant to include more and more dependencies.
Maybe we can find a good way for creating unique names for the SHM files, which doesn't implicate a 7 character topic name length.

2) Let me take a deeper look at the sample. Creating a string with a length from a char array is necessary if that array might contain `\0` characters. I am not sure if that might be the case here, or not.

3) Agreed. 

As mentioned, please make 3 separate PRs for the different issues, they might be easier to track / merge.
Also please make sure to sign the Eclipse CLA as it is mandatory in order to merge your PR.

**Comment by FlorianReimold** [comment]
This was fixed by #1009
